websocket_test.go

```diff
@@ -52,3 +52,286 @@ func TestWebsocketNil(t *testing.T) {
 
 	ws.chain.assertFailed(t)
 }
+
+func TestWebsocketExpect(t *testing.T) {
+
+	t.Run("websocket is closed", func(t *testing.T) {
+		r := newStringReporter()
+		c := &Websocket{
+			chain:    makeChain(r),
+			conn:     &websocket.Conn{},
+			isClosed: true,
+		}
```

Comment:
Here and in tests below. Can we replace manual Websocket construction with invoking NewWebsocket() or makeWebsocket() and probably some other methods when necessary? (e.g. Disconnect to set isClosed to false). Maybe not everywhere, but at least in most places. The less we use private fields in tests, the better and cleaner are tests.

Reply:
Yes, we can replace manual Websocket construction in most cases since isClosed does not need to be set to true. Disconnect function invokes methods on gorilla websocket Conn which cannot be mocked since it is not an interface. Disconnect function requires fully functional gorilla websocket Conn struct.